ofrak_core/ofrak/core/comments.py

```diff
@@ -50,18 +49,34 @@ async def modify(self, resource: Resource, config: AddCommentModifierConfig) ->
 
         # Here I'm appending appending overlapping comments with a new line.
         # Overwriting comments that share a range is counter intuitive and not easily understood without digging into the code.
-        # TODO: Refactor comments strucutre to not key off of the range to allow individual overlapping comment attributes.
-        if config.comment[0] in comments:
-            comment = config.comment[1] + "\n" + comments[config.comment[0]]
-        else:
-            comment = config.comment[1]
-        comments[config.comment[0]] = comment
+        if config.comment[0] not in comments:
+            comments[config.comment[0]] = []
+
+        comments[config.comment[0]].append(config.comment[1])
         resource.add_attributes(CommentsAttributes(comments=comments))
 
 
 @dataclass
 class DeleteCommentModifierConfig(ComponentConfig):
-    comment_range: Optional[Range]
+    """
+    comment_range: Tuple[Optional[Range], comment_text=Optional[str]]
+    If comment_text is provided, deletes the matching comment with the same Optional[Range]
+    If comment_text is None, deletes ALL comments with the same Optional[Range]
+    """
+
+    comment_range: Tuple[Optional[Range], Optional[str]]
```

Comment:
I would separate out the `comment_text` parameter as a separate parameter instead of doing funky stuff with tuples, it doesn't make intuitive sense for the "text" to be part of the "range". Makes the code a lot more readable later on too.

The typing of this param with the processing in `__post_init__` is quite complicated, it would accurately be represented as 

`Union[Optional[Range], Tuple[Optional[Range]], Tuple[Optional[Range], Optional[str]]]`